src/gsy_e/models/strategy/__init__.py

```diff
@@ -428,18 +419,21 @@ def replace(self, old_offer: Offer, new_offer: Offer, market_id: str):
     def on_trade(self, market_id: str, trade: Trade) -> None:
         """Update contents of posted and sold dicts on the event of an offer being traded"""
         try:
-            if trade.seller == self.strategy.owner.name:
-                if trade.offer_bid.id in self.split and trade.offer_bid in self.posted:
-                    # remove from posted as it is traded already
-                    self._remove(self.split[trade.offer_bid.id])
-                self.sold_offer(trade.offer_bid, market_id)
+            if trade.seller.name == self.strategy.owner.name:
```

Comment:
Actually I tried to directly reflect the former code here, however as far as I see this can be fixed because this part of the code should only handle the trades originating from offers and not from bids (event_bid_traded should handle the latter). Fixing